Decoder/DecoderProvider.php

```diff
@@ -11,21 +11,19 @@
 
 namespace FOS\RestBundle\Decoder;
 
-use Symfony\Component\DependencyInjection\ContainerAware;
-
 /**
  * Provides encoders through the Symfony2 DIC.
  *
  * @author Igor Wiedler <igor@wiedler.ch>
  */
-class ContainerDecoderProvider extends ContainerAware implements DecoderProviderInterface
+class DecoderProvider implements DecoderProviderInterface
```

Comment:
this is a good example of a service where I think for performance reasons it would make sense to keep the container injection